### PR TITLE
Bug/dnn 19500

### PR DESCRIPTION
**Pull request by Mohtshum**

### CODE REVIEW

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs [hunk 1]
```diff
@@ -206,24 +206,32 @@ public TabInfo MovePage(PageMoveRequest request)
...
+            var tab = TabController.Instance.GetTab(page.Id, currentPortal.PortalId);
             if (tab == null)
             {
                 throw new PageNotFoundException();
             }
 
             if (TabPermissionController.CanDeletePage(tab))
             {
-                if (TabController.IsSpecialTab(tab.TabID, portalSettings.PortalId))
+                if (TabController.IsSpecialTab(tab.TabID, currentPortal.PortalId))
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs:220]: Are you sure this check should be done only for the current portal and not the actual portal coming from the request? Or both?

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs [hunk 2]
```diff
@@ -206,24 +206,32 @@ public TabInfo MovePage(PageMoveRequest request)
             return TabController.Instance.GetTab(request.PageId, portalSettings.PortalId);
         }
 
-        public void DeletePage(PageItem page)
+        public void DeletePage(PageItem page, PortalSettings portalSettings = null)
         {
-            var portalSettings = PortalController.Instance.GetCurrentPortalSettings();
-            var tab = TabController.Instance.GetTab(page.Id, portalSettings.PortalId);
+            PortalSettings = portalSettings;
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: Is there no side-effect of setting this property?

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs [hunk 3]
```diff
@@ -206,24 +206,32 @@ public TabInfo MovePage(PageMoveRequest request)
...
+                if (TabController.IsSpecialTab(tab.TabID, currentPortal.PortalId))
                 {
                     throw new PageException(Localization.GetString("CannotDeleteSpecialPage"));
                 }
                 else
                 {
-                    TabController.Instance.SoftDeleteTab(tab.TabID, portalSettings);
+                    if (PortalHelper.IsContentExistsForRequestedPortal(tab.PortalID, PortalSettings))
+                    {
+                        TabController.Instance.SoftDeleteTab(tab.TabID, currentPortal);
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: The if uses `PortalSettings` and the delete uses `currentPortal`. Is that correct?

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs (hunk 2 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: Nope this is setting for only DeletePage request path

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs (hunk 1 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs:220]: Actually currentPortal (more intitutive name) is same as protalSettings but it is rename to it as it is getting initialized against same method i.e. PortalController.Instance.GetCurrentPortalSettings();

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs (hunk 3 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: we are only making sure that if portalId is passed as cross-scripting parameter by 

 if (PortalHelper.IsContentExistsForRequestedPortal(tab.PortalID, PortalSettings))

Meanwhile Delete is getting called with same (renamed) parameter.

Comment (tpluscode) [reply src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: I see it's the same thing in all those comments. It's too confusing that there is the property, parameter and local variable.

I understand that all of them should be the same? Please use only the method's params and remove usage of the others from this method.

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs (hunk 3 above)
Comment (Mohtshum) [reply src/Modules/Content/Dnn.PersonaBar.Pages/Components/PagesControllerImpl.cs]: Now have remove the calls for property PortalSettings object and using the parameter object.

**Review by tpluscode — APPROVED** [review]

**Comment by tpluscode** [comment]
fixed #574

**Pull request merged**